src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/ImportExport.cs

```diff
@@ -10,7 +10,8 @@ namespace System.Security.Cryptography.Rsa.Tests
     [SkipOnPlatform(TestPlatforms.Browser, "Not supported on Browser")]
     public partial class ImportExport
     {
-        public static bool Supports16384 { get; } = TestRsa16384();
+        private static readonly Lazy<bool> s_supports16384 = new Lazy<bool>(TestRsa16384);
+        public static bool Supports16384 => s_supports16384.Value;
```

Comment:
TestRsa16384 itself takes a lot of time?

Reply:
~30 seconds.

Reply:
Woh. Ok. That's not something we have any control over, right? That's all time spent in the OS implementation?

Reply:
> That's not something we have any control over, right? That's all time spent in the OS implementation? 
Yeah.

> Why does testing for even the support for this take so long?

It basically boils down to: This is the largest possible RSA key. No one uses them (or should), but it is a theoretical max, so we want to test our upper boundary condition. Using an RSA key that large is a guaranteed way to hit performance problems.

The performance hit is in OpenSSL's `EVP_PKEY_check`. We use this method on key inputs so we can validate the key's consistency (making sure public parameters match private ones, etc). OpenSSL does a rather "thorough" job of this by running Miller-Rabin on the key input. We have no control over that or any flags, and we can't really avoid the called to `EVP_PKEY_check`.

> Woh. Ok.

This PR cut the total execution time of System.Security.Cryptography on Linux by more than half on my Linux box.